### PR TITLE
net/http/authz: return sinkdb Op instead of executing

### PR DESCRIPTION
**Pull request by kr**
This will allow the caller to combine the returned Op
with other Ops into an atomic batch.

### CODE REVIEW

**Comment by kr** [comment]
PTAL

File: net/http/authz/store.go [hunk 1]
```diff
@@ -94,16 +93,12 @@ func (s *Store) Delete(ctx context.Context, policy string, delete func(*Grant) b
 
 	// We didn't match any grants, don't need to do an update. Return success
```
Comment (tessr) [inline net/http/authz/store.go]: Probably should update this comment to "return no-op"

File: database/sinkdb/op.go [hunk 2]
```diff
@@ -69,3 +69,11 @@ func AddAllowedMember(addr string) Op {
 		}},
 	}
 }
+
+// Error returns an Op representing an error condition.
+// Exec will return err, and have no effect,
+// when the returned Op is executed.
+// If err is nil, Error returns the zero Op.
```
Comment (tessr) [inline database/sinkdb/op.go]: I am a leeeettle bit worried that this is non-obvious but maybe it's part of a pattern I'm not familiar with.

Comment (tessr) [inline database/sinkdb/op.go]: Oh, I guess it's similar to Errors.Wrap returning nil if err is nil

File: net/http/authz/store.go [hunk 3]
```diff
@@ -94,16 +93,12 @@ func (s *Store) Delete(ctx context.Context, policy string, delete func(*Grant) b
...
 	}
 
-	gList := &GrantList{Grants: keep}
-	err = s.sdb.Exec(ctx, sinkdb.Set(key, gList))
-	if err != nil {
-		return errors.Wrap(err)
-	}
-
-	return nil
+	// TODO(tessr): Make this safe for concurrent updates. Will likely require a
```
Comment (tessr) [inline net/http/authz/store.go]: does this todo belong here?

Comment (kr) [inline net/http/authz/store.go]: Yes. What if the read-modify-write of Delete interleaves with the read-modify-write of another Delete or a Save?

**Pull request merged**